src/sound_check_menu.c

```diff
@@ -0,0 +1,2198 @@
+    gUnknown_020387D9 = 1;
+    gUnknown_020387B4[CRY_TEST_UNK0] = 0;
+    gUnknown_020387B4[CRY_TEST_VOLUME] = 0x78;
+    gUnknown_020387B4[CRY_TEST_PANPOT] = 0;
+    gUnknown_020387B4[CRY_TEST_PITCH] = 0x3C00;
+    gUnknown_020387B4[CRY_TEST_LENGTH] = 0xB4;
+    gUnknown_020387B4[CRY_TEST_PROGRESS] = 0;
+    gUnknown_020387B4[CRY_TEST_RELEASE] = 0;
+    gUnknown_020387B4[CRY_TEST_CHORUS] = 0;
+    gUnknown_020387B4[CRY_TEST_PRIORITY] = 2;
+    sub_80BAD5C();
+    sub_80BAE10(0, 0);
+    TASK.FUNC = sub_80BAA48;
+}
+
+void sub_80BAA48(u8 taskId) // Task_ProcessDriverTestInput
+{
+    if(gMain.newKeys & 0x2)
+    {
+        REG_DISPCNT = 0x7140;
+        REG_WIN0H = 0x11DF;
+        REG_WIN0V = 0x11F;
+        MenuZeroFillWindowRect(0, 0, 0x1D, 0x13);
+        TASK.FUNC = sub_80BA258;
+        return;
+    }
+    if(gMain.newAndRepeatedKeys & 0x40) // _080BAAA8
+    {
+        u8 backupVar = gUnknown_020387B3;
+        if(--gUnknown_020387B3 < 0)
+            gUnknown_020387B3 = 8;
+
+        sub_80BAE10(backupVar, gUnknown_020387B3);
+        return;
+    }
+    if(gMain.newAndRepeatedKeys & 0x80) // _080BAAD0
+    {
+        u8 backupVar = gUnknown_020387B3;
+        if(++gUnknown_020387B3 > 8)
+            gUnknown_020387B3 = 0;
+
+        sub_80BAE10(backupVar, gUnknown_020387B3);
+        return;
+    }
+    if(gMain.newKeys & 0x8) // _080BAAF8
+    {
+        gUnknown_020387D8 ^= 1;
+        sub_80BAD5C();
+        return;
+    }
+    if(gMain.newKeys & 0x4) // _080BAB14
+    {
+        gUnknown_020387D9 ^= 1;
+        sub_80BAD5C();
+        SetPokemonCryStereo(gUnknown_020387D9);
+        return;
+    }
+    if(gMain.newAndRepeatedKeys & 0x100) // _080BAB38
+    {
+        sub_80BACDC(10);
+        sub_80BAD5C();
+        return;
+    }
+    if(gMain.newAndRepeatedKeys & 0x200) // _080BAB46
+    {
+        sub_80BACDC(-10);
+        sub_80BAD5C();
+        return;
+    }
+    if(gMain.newAndRepeatedKeys & 0x20) // _080BAB56
+    {
+        sub_80BACDC(-1);
+        sub_80BAD5C();
+        return;
+    }
+    if(gMain.newAndRepeatedKeys & 0x10) // _080BAB64
+    {
+        sub_80BACDC(1);
+        sub_80BAD5C();
+        return;
+    }
+    if(gMain.newKeys & 0x1) // _080BAB78
+    {
+        u8 divide, remaining;
+
+        SetPokemonCryVolume(gUnknown_020387B4[CRY_TEST_VOLUME]);
+        SetPokemonCryPanpot(gUnknown_020387B4[CRY_TEST_PANPOT]);
+        SetPokemonCryPitch(gUnknown_020387B4[CRY_TEST_PITCH]);
+        SetPokemonCryLength(gUnknown_020387B4[CRY_TEST_LENGTH]);
+        SetPokemonCryProgress(gUnknown_020387B4[CRY_TEST_PROGRESS]);
+        SetPokemonCryRelease(gUnknown_020387B4[CRY_TEST_RELEASE]);
+        SetPokemonCryChorus(gUnknown_020387B4[CRY_TEST_CHORUS]);
+        SetPokemonCryPriority(gUnknown_020387B4[CRY_TEST_PRIORITY]);
+
+        remaining = gUnknown_020387B4[CRY_TEST_UNK0] % 128;
+        divide = gUnknown_020387B4[CRY_TEST_UNK0] / 128;
+
+        switch(divide)
+        {
+            case 0:
```

Comment:
This is not about habits, but about consistency. I pretty much dislike everything about the codestyle, yet I try to comply.